File: Pythonleetcode/Leetcode134.py
class Solution(object):
    def canCompleteCircuit(self, gas, cost):
        def sucess(i,gas,cost):
            total=0
            mark=0
            while (mark!=len(gas)):

                if i<=len(gas)-1:

                    total=total+gas[i]-cost[i]



                    if total<0:
                        return -10
                    mark=mark+1
                    i=i+1
                else:
                    i = 0
                    total = total + gas[i] - cost[i]


                    if total < 0:
                        return -10
                    mark = mark + 1
                    i=1



            return total




        for i in range(len(gas)):
            if sucess(i,gas,cost)>=0:
                return i
        return -1



print(Solution().canCompleteCircuit([1,2,3,4,5], [3,4,5,1,2]))

print(Solution().canCompleteCircuit( [2,3,4], [3,4,3]))

print(Solution().canCompleteCircuit( [3,3,4],[3,4,4]))

File: CppPractice/CppLeetcode/Leetcode134.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) 
    {
        int ans=0;
        int curGas = 0;
        if(accumulate(gas.begin(),gas.end(),0)-accumulate(cost.begin(),cost.end(),0)<0){return -1;}
        for(int i=0;i<gas.size();i++)
        {
            curGas +=(gas[i]-cost[i]);
            if(curGas<0)
            {
                curGas=0;
                ans=i+1;
            }
        }
        return ans; 
    }
};